dandiapi/api/tasks/__init__.py

```diff
@@ -65,8 +50,9 @@ def write_manifest_files(version_id: int) -> None:
 def validate_asset_metadata_task(asset_id: int) -> None:
     from dandiapi.api.services.metadata import validate_asset_metadata
 
-    asset: Asset = Asset.objects.get(id=asset_id)
-    validate_asset_metadata(asset=asset)
+    asset: Asset = Asset.objects.filter(id=asset_id, status=Asset.Status.PENDING).first()
+    if asset:
+        validate_asset_metadata(asset=asset)
```

Comment:
Correct. This doesn't add any correctness it's strictly an optimization to avoid re-running the entire `validate` method before detecting that there's nothing to update. I'll add a comment.